transcom/mymove
# B 20756

## [B-20756](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A983877)
## [INT Review](https://github.com/transcom/mymove/pull/13970)

## Summary

This task adds functionality for the customer to push ppm shipments through when DTOD is down. They will still recieve error messages for bad zips when DTOD is available.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Prior to running the application, set the DTOD_SIMULATE_OUTAGE to true

1. Access the milmove app
2. Login as a Customer
3. Create a new Move
4. Add a PPM shipment a push it through the process
5. Verify the allowance and advance pages show the correct error messages (example in screenshots at the bottom of this review)
6.  Verify the shipment was created successfully

This process will be the same for editing PPMs as a customer while creating a move. As long as the returned value from the estimated incentive is zero, the estimated allowance and estimated advance will show the messages about DTOD being unnavailable.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-10-23 at 9 22 30 AM](https://github.com/user-attachments/assets/e996fa1b-41bd-400b-b0a7-a244924a8ca0)
![Screenshot 2024-10-23 at 9 22 35 AM](https://github.com/user-attachments/assets/7880c718-e5bd-44f3-b3df-a21919c8dcd0)
